### ISSUE TITLE
[Bug]: Rexported symbols like `sqrt` cause issues when imported using `.{..}`

### ISSUE THREAD

**Issue by ShreyasKhandekar** [chapel-lang/chapel#25172]
### Summary of Problem

**Description:**
Symbols like `sqrt` which are in `Automath` but are re-exported by other modules like `Math` cause issues when imported in the form`import Math.{sqrt};`

The error seen is as follows:

```console
sqrt.chpl:13: error: Bad identifier, no known 'sqrt' defined in 'Math'
```

**Is this issue currently blocking your progress?**
no, the issue is worked around by doing `import Math.sqrt;` instead.

### Steps to Reproduce

**Source Code:**

```chapel
import Math.{sqrt};
```

**Compile command:**
`chpl sqrt.chpl`

**Execution command:**
`./sqrt`

**Associated Future Test(s):**
none

### Configuration Information

<details>
<summary> Reveal </summary>
 
- Output of `chpl --version`:
```
chpl version 2.1.0 pre-release (7e181b1d05)
  built with LLVM version 15.0.4
  available LLVM targets: nvptx64, nvptx, x86-64, x86, amdgcn, r600
Copyright 2020-2024 Hewlett Packard Enterprise Development LP
Copyright 2004-2019 Cray Inc.
(See LICENSE file for more details)
```
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:
```
CHPL_TARGET_PLATFORM: linux64
CHPL_TARGET_COMPILER: llvm
CHPL_TARGET_ARCH: x86_64
CHPL_TARGET_CPU: none *
CHPL_LOCALE_MODEL: flat
CHPL_COMM: gasnet *
  CHPL_COMM_SUBSTRATE: udp
  CHPL_GASNET_SEGMENT: fast *
CHPL_TASKS: qthreads
CHPL_LAUNCHER: amudprun
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: cstdlib
  CHPL_NETWORK_ATOMICS: none
CHPL_GMP: bundled
CHPL_HWLOC: bundled
CHPL_RE2: bundled
CHPL_LLVM: system
CHPL_AUX_FILESYS: none
```

-- `gcc --version`
```
gcc (SUSE Linux) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

</details>

**Comment by ShreyasKhandekar** [comment]
I'm not sure if the re-exporting by itself is the issue. Maybe the reason also involves the fact the Automath is automatically included.
For example, the following is NOT a reproducer even though it uses the same pattern:
```chapel
module C {
  proc foo() {
    writeln("Just chilling");
  }
}

module B {
  public use C;
}

module A {
  proc main() {
    use C;
    import B.{foo};
    foo();
  }
}

```

**Comment by bradcray** [comment]
@ShreyasKhandekar : Interesting observation, and good thought to try to reproduce it outside of the module system.  It'd be interesting to see if adding a line to ChapelStandard.chpl similar to/next to the AutoMath line for your module C caused your program to behave differently.

**Comment by DanilaFe** [comment]
In case anyone else was curious, I tried compiling the above program with `--no-dyno-scope-resolve`, and still got the error (it's probably coming from the production scope resolve pass). By a similar idea, I compiled with `--dyno-scope-bundled,` and still got the error. If anything, it looks like this is something that the dyno scope resolve is not handling at all.

Also interestingly, I was able to use the (Dyno-based) LSP to correctly identify `sqrt`, and `testInteractive` did not report any errors.
<img width="514" alt="Screen Shot 2024-06-04 at 8 52 46 PM" src="https://github.com/chapel-lang/chapel/assets/4361282/21653a84-a7a9-4b21-bcb5-1c5df8f21dd3">
